workflows/data_io_utils/file_rules/nodes.py

```diff
@@ -61,6 +69,87 @@ class Directory(File):
         repr=False,
     )
 
+    @field_validator("files", mode="before")
+    @classmethod
+    def validate_files(cls, files_input, info):
+        """
+        Validate and transform the files input.
+
+        This validator handles different types of file specifications:
+        - File objects (used as-is)
+        - Directory objects (used as-is)
+        - Strings (converted to File objects with default rules)
+        - Tuples of (filename, rules) where filename is a string and rules is a list of FileRule
+
```

Comment:
Might it be more pydantic-native to, instead, have this just coerce string or tuple inputs into File objects that are then set to the `files` attribute? This feels like it perhaps hides a bit too much logic. 

As a second-best option, if we could at least specify a type hint for shape of the tuple input that would be good